search/house_numbers_matcher.cpp

```diff
@@ -545,17 +549,23 @@ bool HouseNumbersMatch(strings::UniString const & houseNumber, vector<Token> con
   if (houseNumber.empty() || queryParse.empty())
     return false;
 
-  // Fast pre-check, helps to early exit without complex house number
-  // parsing.
-  if (IsASCIIDigit(houseNumber[0]) && IsASCIIDigit(queryParse[0].m_value[0]) &&
-      houseNumber[0] != queryParse[0].m_value[0])
-  {
-    return false;
+  vector<Token> tokens;
+  Tokenize(houseNumber, false, tokens);
```

Comment:
Needs to verify with benchmarks,  because of
// Fast pre-check, helps to early exit without complex house number parsing.
Seems like this function is very sensitive for speed.